### ISSUE TITLE
examples/matplotlib/pcolor.vt fails on matplotlib 1.4.3

### ISSUE THREAD

**Issue by rexissimus** [VisTrails/VisTrails#1132]
It looks like the arguments to pcolor are now stricter but I am not sure how to create valid input data.

```
  errorTrace: Traceback (most recent call last):
  File "C:\Program Files\VisTrails x64 gdal\vistrails\core\modules\vistrails_module.py", line 573, in update
    self.compute()
  File "C:\Program Files\VisTrails x64 gdal\vistrails\packages\matplotlib\bases.py", line 107, in compute
    plot(self.figInstance)
  File "C:\Program Files\VisTrails x64 gdal\vistrails\packages\matplotlib\plots.py", line 3037, in <lambda>
    args, kwargs))
  File "C:\Program Files\VisTrails x64 gdal\vistrails\packages\matplotlib\plots.py", line 3040, in plot_figure
    polyCollection = matplotlib.pyplot.pcolor(*args, **kwargs)
  File "C:\Program Files\VisTrails x64 gdal\Python27_64\lib\site-packages\matplotlib\pyplot.py", line 3018, in pcolor
    ret = ax.pcolor(*args, **kwargs)
  File "C:\Program Files\VisTrails x64 gdal\Python27_64\lib\site-packages\matplotlib\axes\_axes.py", line 4889, in pcolor
    X, Y, C = self._pcolorargs('pcolor', *args, allmatch=False)
  File "C:\Program Files\VisTrails x64 gdal\Python27_64\lib\site-packages\matplotlib\axes\_axes.py", line 4692, in _pcolorargs
    numRows, numCols = C.shape
AttributeError: 'list' object has no attribute 'shape'
```

**Comment by tacaswell** [comment]
This might be a regression on the mpl side.

I have a vague memory of something about this going by, can you check if it works on master?

**Comment by rexissimus** [comment]
It does not work on mpl master. This is the code used to create the input to pcolor(X, Y, Z):
```
import numpy as np

def func3(x,y):
    return (1 - x/2 + x**5 + y**3) * np.exp(-x**2 - y**2)


# make these smaller to increase the resolution
dx, dy = 0.05, 0.05

x = np.arange(-3.0, 3.0001, dx)
y = np.arange(-3.0, 3.0001, dy)
X, Y = np.meshgrid(x, y)

Z = func3(X, Y)
```

Googling the problem turns up a few suggestions, but nothing that fixes the problem.

**Comment by tacaswell** [comment]
This works correctly.

```python

import matplotlib.pyplot as plt
import numpy as np

def func3(x, ):
    return (1 - x/2 + x**5 + y**3) * np.exp(-x**2 - y**2)


# make these smaller to increase the resolution
dx, dy = 0.05, 0.05

x = np.arange(-3.0, 3.0001, dx)
y = np.arange(-3.0, 3.0001, dy)
X, Y = np.meshgrid(x, y)

Z = func3(X, Y)

plt.pcolor(X, Y, Z)
```

I suspect that you can fix this on the vt side by calling `np.asarray` or `np.vstack` on the data before passing it off to `pcolor`. I have a rough recollection that vt passes arrays between nodes as lists?

**Comment by dakoop** [comment]
It looks like the issue may be some list handling on the VT side.

**Comment by rexissimus** [comment]
Fixed by 11db25b3f6be66c877c6c7705cf9e2ebc1bf5d32. VisTrails converted the array into a list when reading module inputs. This was both bad for performance and could cause errors when sending data other than lists on List ports.

This problem still exists if you pass arrays on 2 connections to a single List port. But there is no reason you would expect that to work anyway. 

This issue was introduced when list handling was updated. It has probably never been possible to use a lists as the C argument to pcolor.

**Comment by tacaswell** [comment]
Well, we also have a PR open _to_ make it work https://github.com/matplotlib/matplotlib/pull/5206
